### ISSUE TITLE
R2RML & Oracle DB: mapping to Datetime loses time information

### ISSUE THREAD

**Issue by yoadey** [ontop/ontop#296]
### Description

We have an R2RML file, which should transform data from an Oracle DB to RDF format.
In Oracle DB, there is a column with type DATE, which also stores the time information. When we define an object with type xsd:dateTime, the time information is lost, only the date (e.g. 2019-08-08) remains.

### Steps to Reproduce

1. Create an Oracle table with an ID and a column of type DATE
```
CREATE TABLE oracle_date_issue (
   id        NUMBER(5) PRIMARY KEY,
   created   DATE DEFAULT (sysdate)
)
```
2. Create an R2RML file which requests data from this table:
```
@prefix rr: <http://www.w3.org/ns/r2rml#> .
@prefix map: <http://github.com/ontop/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix dct: <http://purl.org/dc/terms/> .
map:ORACLE_DATE_ISSUE a rr:TriplesMap;
  rr:logicalTable [ rr:tableName "ORACLE_DATE_ISSUE"; ];
  rr:subjectMap [
    rr:template "http://github.com/ontop/oracledate/{ID}";
  ];
  rr:predicateObjectMap [
    rr:predicate dct:created;
    rr:objectMap [
      rr:column "CREATED";
      rr:datatype xsd:date;
    ];
  ];
.
```
3. Run the r2rml file against the Oracle DB
```
Builder<? extends Builder> builder = OntopStandaloneSQLConfiguration.defaultBuilder();
builder.enableDefaultDatatypeInference(true)//
        .r2rmlMappingGraph(graph) //
        .jdbcUser(user) //
        .jdbcPassword(password) //
        .jdbcUrl(url) //
        .jdbcDriver("oracle.jdbc.driver.OracleDriver") //
        .enableTestMode();
OntopStandaloneSQLConfiguration ontopConfig = builder.build();
InputQueryFactory queryFactory = ontopConfig.getInputQueryFactory();
OntopQueryEngine queryEngine = ontopConfig.loadQueryEngine();
OntopConnection conn = queryEngine.getConnection();
OntopStatement st = conn.createStatement();
SelectQuery query = queryFactory.createSelectQuery("construct { ?s ?p ?o } where { ?s ?p ? o}");
TupleResultSet rs = st.execute(query);
OntopBindingSet bindingSet = rs.next();
Object value = bindingSet.getValue();
System.out.println(value);
```

**Expected behavior:**
The date with time information.

**Actual behavior:**
Only date information is printed

**Reproduces how often:** always

### Additional Information

I checked the code and the problematic part seems to be in OracleSQLDialectAdapter. In line 49 is defined, that TIMESTAMP should be cast to VARCHAR(4000). During this cast the time information is lost. If I replace it with DATE, I don't have any problems.

### Versions

Current master branch as of 2019-08-06

**Comment by ghxiao** [comment]
Hi @yoadey , it seems that you already found a solution. Would you like to make a pull request?

**Comment by guerda** [comment]
I missed the notification on the merge request by yoaday. @ghxiao : Would it be possible to check out his merge request? Thanks!

**Comment by bcogrel** [comment]
Fixed by a PR (#305) merged into the `version4` branch.
